metals/src/main/scala/scala/meta/internal/metals/debug/SetBreakpointsRequestHandler.scala

```diff
@@ -0,0 +1,88 @@
+package scala.meta.internal.metals.debug
+import org.eclipse.lsp4j.debug.SetBreakpointsArguments
+import org.eclipse.lsp4j.debug.SetBreakpointsResponse
+import org.eclipse.lsp4j.debug.Source
+import org.eclipse.lsp4j.debug.SourceBreakpoint
+import org.eclipse.lsp4j.jsonrpc.messages.ResponseMessage
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.meta.internal.metals.JvmSignatures
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.mtags.Mtags
+import scala.meta.internal.semanticdb.SymbolOccurrence
+
+private[debug] final class SetBreakpointsRequestHandler(
+    server: ServerAdapter,
+    adapters: MetalsDebugAdapters
+)(implicit ec: ExecutionContext) {
+
+  // TODO the partitioning does not work for cases when the same file has other types defined
```

Comment:
I'm not sure I understand this TODO, could we open a followup ticket?

Reply:
@marek1840 can we link to the issue here?

Reply:
done!